diff --git a/embuilder.py b/embuilder.py
index cbbb15b..559d14c 100755
--- a/embuilder.py
+++ b/embuilder.py
@@ -66,6 +66,7 @@ PORT_VARIANTS = {
     'sdl2-mt': ('sdl2', {'USE_PTHREADS': 1}),
     'icu-mt': ('icu', {'USE_PTHREADS': 1}),
     'sdl2_mixer_mp3': ('sdl2_mixer', {'SDL2_MIXER_FORMATS': ["mp3"]}),
+    'sdl2_mixer_mid_mp3_ogg': ('sdl2_mixer', {'SDL2_MIXER_FORMATS': ["mid", "mp3", "ogg"]}),
     'sdl2_mixer_none': ('sdl2_mixer', {'SDL2_MIXER_FORMATS': []}),
     'sdl2_image_png': ('sdl2_image', {'SDL2_IMAGE_FORMATS': ["png"]}),
     'sdl2_image_jpg': ('sdl2_image', {'SDL2_IMAGE_FORMATS': ["jpg"]}),
diff --git a/tools/ports/sdl2_mixer.py b/tools/ports/sdl2_mixer.py
index ea943ff..0c9dbab 100644
--- a/tools/ports/sdl2_mixer.py
+++ b/tools/ports/sdl2_mixer.py
@@ -7,8 +7,8 @@ import os
 import shutil
 import logging
 
-TAG = 'release-2.0.2'
-HASH = 'b9d03061d177f20f4e03f3e3553afd7bfe0c05da7b9a774312b389318e747cf9724e0475e9afff6a64ce31bab0217e2afb2619d75556753fbbb6ecafa9775219'
+TAG = 'release-2.0.4'
+HASH = '5ba387f997219a1deda868f380bf7ee8bc0842261dd54772ad2d560f5282fcbe7bc130e8d16dccc259eeb8cda993a0f34cd3be103fc38f8c6a68428a10e5db4c'
 
 deps = ['sdl2']
 
@@ -17,11 +17,7 @@ def needed(settings):
   return settings.USE_SDL_MIXER == 2
 
 
-def get(ports, settings, shared):
-  sdl_build = os.path.join(ports.get_build_dir(), 'sdl2')
-  assert os.path.exists(sdl_build), 'You must use SDL2 to use SDL2_mixer'
-  ports.fetch_project('sdl2_mixer', 'https://github.com/emscripten-ports/SDL2_mixer/archive/' + TAG + '.zip', 'SDL2_mixer-' + TAG, sha512hash=HASH)
-
+def get_lib_name(settings):
   settings.SDL2_MIXER_FORMATS.sort()
   formats = '-'.join(settings.SDL2_MIXER_FORMATS)
 
@@ -30,10 +26,19 @@ def get(ports, settings, shared):
     libname += '_' + formats
   libname += '.a'
 
+  return libname
+
+
+def get(ports, settings, shared):
+  sdl_build = os.path.join(ports.get_build_dir(), 'sdl2')
+  assert os.path.exists(sdl_build), 'You must use SDL2 to use SDL2_mixer'
+  ports.fetch_project('sdl2_mixer', 'https://github.com/libsdl-org/SDL_mixer/archive/' + TAG + '.zip', 'SDL2_mixer-' + TAG, sha512hash=HASH)
+  libname = get_lib_name(settings)
+
   def create(final):
     logging.info('building port: sdl2_mixer')
 
-    source_path = os.path.join(ports.get_dir(), 'sdl2_mixer', 'SDL2_mixer-' + TAG)
+    source_path = os.path.join(ports.get_dir(), 'sdl2_mixer', 'SDL_mixer-' + TAG)
     dest_path = os.path.join(ports.get_build_dir(), 'sdl2_mixer')
 
     shutil.rmtree(dest_path, ignore_errors=True)
@@ -90,7 +95,8 @@ def get(ports, settings, shared):
 
 
 def clear(ports, settings, shared):
-  shared.Cache.erase_lib('libSDL2_mixer.a')
+  libname = get_lib_name(settings)
+  shared.Cache.erase_lib(libname)
 
 
 def process_dependencies(settings):
