diff --git a/addons/audio/sdl_audio.c b/addons/audio/sdl_audio.c
index 777324259..636d7d381 100644
--- a/addons/audio/sdl_audio.c
+++ b/addons/audio/sdl_audio.c
@@ -76,8 +76,10 @@ static int sdl_allocate_voice(ALLEGRO_VOICE *voice)
    want.freq = voice->frequency;
    want.format = allegro_format_to_sdl(voice->depth);
    want.channels = al_get_channel_count(voice->chan_conf);
+   // The a5 SDL audio system hardcodes a value for # samples that is too high. Until I can upstream a patch to make this
+   // configurable, just manually change it here!
    // TODO: Should make this configurable somehow
-   want.samples = 4096;
+   want.samples = 512;
    want.callback = audio_callback;
    want.userdata = sv;
 
diff --git a/src/sdl/sdl_display.c b/src/sdl/sdl_display.c
index 12fc8f14d..afbe4703d 100644
--- a/src/sdl/sdl_display.c
+++ b/src/sdl/sdl_display.c
@@ -321,7 +321,7 @@ static void sdl_flip_display(ALLEGRO_DISPLAY *d)
    SDL_GL_SwapWindow(sdl->window);
 
    // SDL loses texture contents, for example on resize.
-   al_backup_dirty_bitmaps(d);
+   // al_backup_dirty_bitmaps(d);
 }
 
 static void sdl_update_display_region(ALLEGRO_DISPLAY *d, int x, int y,
diff --git a/src/sdl/sdl_system.c b/src/sdl/sdl_system.c
index ee7034488..f859e9177 100644
--- a/src/sdl/sdl_system.c
+++ b/src/sdl/sdl_system.c
@@ -132,7 +132,8 @@ static void sdl_heartbeat(void)
 #ifdef __EMSCRIPTEN__
    double t = al_get_time();
    double interval = t - s->timer_time;
-   _al_timer_thread_handle_tick(interval);
+   // This emscripten-specific timer code actually really messes up the framerate, making it go way too fast.
+   // _al_timer_thread_handle_tick(interval);
    s->timer_time = t;
 #endif
    al_unlock_mutex(s->mutex);
